### PR TITLE
Maximum numbers of connected ESCs in uORB and in UAVCAN expanded to 16.

### PR DESCRIPTION
**Pull request by irsdkv**
**Describe problem solved by the proposed pull request**
At this time maximum number of supported ESC is 8.
This restriction does not meet modern requirements.
For example, some VTOLs could have 8 motors for MC mode and one or more push/pull motors for FW mode.

**Test data / coverage**
It was tested on real VTOL with 8 MC motors and one push FW motor (with Pixhawk 4 onboard).
The flying plan included MC mode and FW mode.
The flight went well, without any lags or problems.

**Describe your preferred solution**
Increase maximum number of connected ESC in uORB messages and in UAVCAN ESC driver.

**Additional context**
This [PR](https://github.com/PX4/Firmware/pull/12166) also associated with this task as it allow to use VTOL copters with more than 8 motors in simulator.

### CODE REVIEW

**Comment by dagar** [comment]
One minor concern is the size of `esc_status`. We should probably restructure this at some point to publish the individual status messages with identifiable information.

I'd love to see a log if you're able to share it (logs.px4.io).

File: src/drivers/uavcan/uavcan_main.cpp [hunk 1]
```diff
@@ -901,9 +901,11 @@ int UavcanNode::run()
 
 			} else if (controls_updated && (_mixers != nullptr)) {
 
-				// XXX one output group has 8 outputs max,
+				// XXX one output group has 16 (esc_status_s::CONNECTED_ESC_MAX) outputs max (Q2/2019),
 				// but this driver could well serve multiple groups.
-				unsigned num_outputs_max = 8;
+				unsigned num_outputs_max = (
```
Comment (dagar) [inline src/drivers/uavcan/uavcan_main.cpp]: Why is this necessary?

Comment (irsdkv) [reply src/drivers/uavcan/uavcan_main.cpp]: You are right. I'm not sure about the need for this check.
Perhaps the best solution would be to assign the constant value 16 (esc_status_s::CONNECTED_ESC_MAX).
But, on the other hand, I did not want to bind to a constant, which in the distant future could still change.
If you think that the constant value would be better, I will be glad to change this.

Comment (irsdkv) [reply src/drivers/uavcan/uavcan_main.cpp]: I think that the constexpr specifier solve all problems with CPU load and this check (CONNECTED_ESC_MAX constant presence) will be useful for further changes safety.

**Comment by irsdkv** [comment]
> I'd love to see a log if you're able to share it (logs.px4.io).

Sure! Next week we will provide the logs.

**Comment by LorenzMeier** [comment]
@irsdkv Could you please share the logs?

**Comment by irsdkv** [comment]
> @irsdkv Could you please share the logs?

Sorry for delay! Our pilot is on vacation now.  We will provide logs as fast as we can. I suppose that logs will be available at the beginning of July.

**Comment by julianoes** [comment]
What's the status here? @irsdkv is this still something you would like to get in or should it be closed?

**Comment by irsdkv** [comment]
I would like to get it, but we still have not pilot and unfortunately at this time we can't provide the logs.

Arch was changed after creating this PR so I think we should close it. 

I will create new request when we will ready.

**Comment by irsdkv** [comment]
I would like to get it, but we still have not pilot and unfortunately at this time we can't provide the logs.

Arch was changed after creating this PR so I think we should close it.

I will create new request when we will ready.
